lib/pr-triage.js

```diff
@@ -82,12 +87,62 @@ class PRTriage {
 
       if (changeRequestedReviews.length > 0) {
         return PRTriage.STATE.CHANGES_REQUESTED;
+      } else if (
+        approvedReviews.length < numNeededReviews ||
+        numRequestedReviewsRemaining > 0
+      ) {
+        // Mark if partially approved if:
+        // 1) Branch protections require more approvals
+        //  - or -
+        // 2) not everyone requested has approved (requested remaining > 0)
+        return PRTriage.STATE.PARTIALLY_APPROVED;
       } else if (reviews.length === approvedReviews.length) {
         return PRTriage.STATE.APPROVED;
       }
     }
   }
 
+  /**
+   * Get the number of required number of reviews according to branch protections
+   * @return {Promise<number>} The number of required approving reviews, or 1 if branch protections are not set up.
+   * @private
+   */
+  async _getRequiredNumberOfReviews() {
+    const { owner, repo } = this.config;
+    const branch = this.pullRequest.base.ref;
+    const branchProtections = await this.github.repos.getBranchProtection({
+      owner,
+      repo,
+      branch
+    });
+
+    const requiredReviews =
+      branchProtections.data.required_pull_request_reviews;
+    return (
+      (requiredReviews && requiredReviews.required_approving_review_count) || 1
+    );
+  }
+
+  /**
+   * Get the number of users and teams that have been requested to review the PR
+   * @return {Promise<number>}
+   * @private
+   */
+  async _getRequestedNumberOfReviews() {
+    const { owner, repo } = this.config;
+    const pull_number = this.pullRequest.number;
```

Comment:
It would be good if it is camelCase. 

I understand there are variable names with snake case. After merging this pull request, I will fix these other issues.